client-course-schedulizer/src/components/Tabs/FacultyLoads/FacultyLoads.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from "@material-ui/core";
+import React, { useContext, useMemo } from "react";
+import { Column, useTable } from "react-table";
+import { Schedule, Term } from "../../../utilities/interfaces/dataInterfaces";
+import { ScheduleContext } from "../../../utilities/services/context";
+
+interface TableData {
+  faculty: string;
+  fallCourseSections?: string;
+  fallHours?: number;
+  loadNotes?: string;
+  otherDuties?: string;
+  otherHours?: number;
+  springCourseSections?: string;
+  springHours?: number;
+  summerCourseSections?: string;
+  summerHours?: number;
+  totalHours?: number;
+}
+
+const createTable = (schedule: Schedule): TableData[] => {
+  const newTableData: TableData[] = [];
+  schedule.courses.forEach((course) => {
+    course.sections.forEach((section) => {
+      section.instructors.forEach((instructor) => {
+        const instructorFullName = `${instructor.firstName} ${instructor.lastName}`;
+        const sectionName = `${course.prefixes[0]}-${course.number}-${section.letter}`;
+        const newDataRow: TableData = {
+          faculty: instructorFullName,
+        };
+        const [oldDataRow] = newTableData.filter((data) => {
+          return data.faculty === instructorFullName;
+        });
+        switch (section.term) {
```

Comment:
Better way for this switch? Lots of similar repetitions but I'm not if it can be better. Maybe the cases should stay but turn this into a function. If you find a bug in one, then you would have to fix it in all. 
```
            newDataRow.fallCourseSections = oldDataRow?.fallCourseSections
              ? `${oldDataRow.fallCourseSections}, ${sectionName}`
              : sectionName;
            newDataRow.fallHours = oldDataRow?.fallHours
              ? oldDataRow.fallHours + (section.facultyHours || course.facultyHours)
              : section.facultyHours || course.facultyHours;
```

Reply:
Yeah I thought about doing that but decided that it may be confusing with the number of parameters that need to be passed, and assignment in a different function is kind of weird, but you're right that it's probably better since it'd prevent the duplicated code so I'll do my best to fix that